src/lib/DataSourceDefinition/Base.ts

```diff
@@ -7,20 +7,37 @@ export default abstract class Base {
   static get label(): string {
     throw new Error("Not Implemented");
   }
+  static get configSchema(): ConfigSchemaType {
+    throw new Error("Not Implemented");
+  }
 
   constructor(config) {
     this.config = config;
   }
 
-  abstract execute(query);
+  abstract execute(query: string): Promise<any>;
 
-  abstract cancel();
+  // @todo Set return type as Promise<void> ?
+  abstract cancel(): void | Promise<void>;
 
-  abstract connectionTest();
+  // @todo Define type of the result (boolean or void ?)
+  abstract connectionTest(): Promise<any>;
```

Comment:
Datasources have different return value type.
(true or void)
I suggest to use `Promise<void>`.